plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/common/ui/wizard/AbstractApplicationWorkbenchWizard.java

```diff
@@ -65,20 +65,18 @@ public boolean canFlipToNextPage() {
 		@Override
 		public IWizardPage getNextPage() {
 			IWizardPage page = null;
-			if (isConnected()) {
-				if (wizard != null) {
-					page = wizard.getStartingPage();
-				}
-			} else if (DataBindingUtils.isValid(getDatabindingContext())
-					&& connect()) {
-				IConnection connection = getConnection();
-				this.wizard = getWizard(connection);
-				if (wizard != null) {
-					wizard.setConnection(connection);
-					page = wizard.getStartingPage();
+			if (isConnected() && wizard != null) {
+				page = wizard.getStartingPage();
+			} else if (DataBindingUtils.isValid(getDatabindingContext())) {
```

Comment:
I somehow didn't like the connect() (which does some action) in the 2nd half of an else statement, where I would typically be expecting checks on some boolean or some value comparison.  Basically, I thought "connect()" was hidden in the 2nd half and deserved a more prominent location since it actually takes some action.